firebase-query.html

```diff
@@ -309,17 +322,45 @@
           return -1;
         },
 
+        __onFirebaseValue: function(snapshot) {
+          this.query.off('value', this.__onFirebaseValue, this);
+
+          if (snapshot.hasChildren()) {
+            var items = snapshot.val();
+            var keys = Object.keys(items);
+            var data = [];
+
+            for (var i = 0, len = keys.length; i < len; i++) {
+              var key = keys[i];
+              var value = this.__valueWithKey(key, items[key]);
+
+              this._log('Firebase value added:', key, value);
+
+              this.__map[key] = value;
+              data.push(value);
+            }
+
+            this.set('data', data);
+          }
+
+          /* Now start listening for changes */
+          this.query.on('child_added', this.__onFirebaseChildAdded, this.__onError, this);
```

Comment:
@mikelehen I forked @dickp 's branch and asked for a PR on his side that is connected to here, using what you have suggested

https://github.com/dickp/polymerfire/pull/1